src/NUnitFramework/framework/Constraints/ConstraintBuilder.cs

```diff
@@ -34,14 +34,14 @@ namespace NUnit.Framework.Constraints
     /// input constraints as well as the results of each
     /// operator applied.
     /// </summary>
-    public class ConstraintBuilder : IResolveConstraint
+    public sealed class ConstraintBuilder : IResolveConstraint
```

Comment:
As far as I can tell, there is no useful way to inherit from this class. If a class is not specifically designed with inheritance in mind, we should signal that like we do with readonly on fields. Inheritance comes with often-unintended consequences. (A simple example is that straight-forward custom equality comparisons stop being symmetric).

Reply:
Sounds ok to me